tokenize_rt.py

```diff
@@ -6,35 +6,65 @@
 import io
 import re
 import tokenize
+from typing import Iterable
+from typing import Iterator
+from typing import List
+from typing import Optional
+from typing import Pattern
+from typing import Sequence
+from typing import Tuple
+
+if False:
+    from typing import Text
 
 
 ESCAPED_NL = 'ESCAPED_NL'
 UNIMPORTANT_WS = 'UNIMPORTANT_WS'
 NON_CODING_TOKENS = frozenset(('COMMENT', ESCAPED_NL, 'NL', UNIMPORTANT_WS))
-Offset = collections.namedtuple('Offset', ('line', 'utf8_byte_offset'))
-Offset.__new__.__defaults__ = (None, None)
-Token = collections.namedtuple(
+
+
+class Offset(collections.namedtuple('Offset', ('line', 'utf8_byte_offset'))):
+    __slots__ = ()
+
+    def __new__(cls, line=None, utf8_byte_offset=None):
+        # type: (Optional[int], Optional[int]) -> None
+        return super(Offset, cls).__new__(cls, line, utf8_byte_offset)
```

Comment:
bleh, this is kinda the worst part :/

Reply:
Yea happy to hold off until `defaults=` is backported from 3.7 and mypy understands it :/

Reply:
(or something else)

Reply:
Any ideas for tests to make sure this didn't regress?

Reply:
this is fine I guess 🤷‍♂